upstairs/src/volume.rs

```diff
@@ -1238,25 +1246,167 @@ impl Volume {
                 "Only a single sub_volume is supported"
             )
         }
-        // Sub volumes must all be VolumeConstructionRequest::Region
+
+        // We support replacement for read_only_parent targets with the
+        // same rules as sub_volume targets, however, for any single VCR
+        // update, only one of the sub_volume or read_only_parent targets
+        // can change.
+        let rop_result = match Self::compare_vcr_region_for_replacement(
+            o_read_only_parent,
+            n_read_only_parent,
+        ) {
+            Err(e) => {
+                warn!(log, "read_only_parent error: {}", e);
+                return Err(e);
+            }
+            Ok(rr) => rr,
+        };
+
+        // Determine the sub_volume difference if there is any.
```

Comment:
Add an assertion that `o_sub_volumes.len() == 1` and `n_sub_volumes.len() == 1`?

Reply:
We do check that the sub_volume lengths are 1 above. (line ~1244).